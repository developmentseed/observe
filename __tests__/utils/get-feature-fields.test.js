/* global fetch */
/* global test */
/* global __dirname */
/* global expect */

import getFeatureFields from '../../app/utils/get-feature-fields'
import { getFeature } from '../test-utils'

test('get fields of a feature', () => {
  const featureWithFields = getFeature({ 'properties': {
    'id': 'node/observe-vfxnxtmo20j',
    'version': 1,
    'name': 'Test',
    'building': 'house',
    'icon': 'maki_marker'
  } })
  const featureWithoutFields = getFeature()

  expect(getFeatureFields(featureWithFields)).toMatchSnapshot()
  expect(getFeatureFields(featureWithoutFields)).toMatchSnapshot()
})
