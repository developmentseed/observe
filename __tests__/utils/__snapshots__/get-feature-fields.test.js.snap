// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get fields of a feature 1`] = `
Array [
  Object {
    "key": "id",
    "value": Object {
      "properties": Object {
        "building": "house",
        "icon": "maki_marker",
        "id": "node/observe-vfxnxtmo20j",
        "name": "Test",
        "version": 1,
      },
    },
  },
]
`;

exports[`get fields of a feature 2`] = `
Array [
  Object {
    "key": "id",
    "value": undefined,
  },
]
`;
